Solutions/Ais.Net.Receiver/ReceiverHost.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="Program.cs" company="Endjin">
+// Copyright (c) Endjin. All rights reserved.
+// </copyright>
+
+namespace Ais.Net.Receiver
+{
+    using Microsoft.Extensions.Configuration;
+
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reactive.Linq;
+    using System.Threading.Tasks;
+
+    public class ReceiverHost
+    {
+        private readonly IConfiguration configuration;
+        private readonly IStorageClient storageClient;
+        private NmeaReceiver receiver;
+
+        public ReceiverHost(IConfiguration configuration, IStorageClient storageClient)
+        {
+            this.configuration = configuration;
+            this.storageClient = storageClient;
+        }
+
+        public async Task StartAsync()
+        {
+            this.receiver = new NmeaReceiver(
+                this.configuration[ConfigurationKeys.Host],
+                int.Parse(this.configuration[ConfigurationKeys.HostPort]),
+                TimeSpan.Parse(this.configuration[ConfigurationKeys.ConnectionRetryPeriodicity]),
+                int.Parse(this.configuration[ConfigurationKeys.ConnectionRetryAttempts]));
+
+            IObservable<string> messages = this.GetAsync().ToObservable();
```

Comment:
So having gone to great trouble to replace Rx with `IAsyncEnumerable<T>` we finally turn it back into Rx again here. Why did we need to convert everything else to `IAsyncEnumerable<T>`?